tests/cpp/unit-tests/test_redis.cpp

```diff
@@ -0,0 +1,303 @@
+            THEN("Default member variable values are used")
+            {
+                check_all_defaults(redis_server);
+            }
+        }
+        else {
+            RedisTest redis_server;
+            THEN("Default member variable values are used")
+            {
+                check_all_defaults(redis_server);
+            }
+        }
+    }
+    GIVEN("A Redis derived object with valid environment variables set")
+    {
+        int conn_timeout = 5; //seconds
+        int conn_interval = 500; //milliseconds
+        int cmd_timeout = 2; //seconds
+        int cmd_interval = 250; //milliseconds
+        int expected_conn_attempts = 11;
+        int expected_cmd_attempts = 9;
+
+        unset_all_env_vars();
+        setenv(CONN_TIMEOUT_ENV_VAR, std::to_string(conn_timeout).c_str(), true);
+        setenv(CONN_INTERVAL_ENV_VAR, std::to_string(conn_interval).c_str(), true);
+        setenv(CMD_TIMEOUT_ENV_VAR, std::to_string(cmd_timeout).c_str(), true);
+        setenv(CMD_INTERVAL_ENV_VAR, std::to_string(cmd_interval).c_str(), true);
+
+        if (use_cluster()) {
+            RedisClusterTest redis_server;
+            THEN("Environment variables are used for member variables")
+            {
+                CHECK(redis_server.get_connection_timeout() ==
+                      conn_timeout);
+                CHECK(redis_server.get_connection_interval() ==
+                      conn_interval);
+                CHECK(redis_server.get_connection_attempts() ==
+                      expected_conn_attempts);
+
+                CHECK(redis_server.get_command_timeout() ==
+                      cmd_timeout);
+                CHECK(redis_server.get_command_interval() ==
+                      cmd_interval);
+                CHECK(redis_server.get_command_attempts() ==
+                      expected_cmd_attempts);
+            }
+        }
+        else {
+            RedisTest redis_server;
+            THEN("Environment variables are used for member variables")
+            {
+                CHECK(redis_server.get_connection_timeout() ==
+                      conn_timeout);
+                CHECK(redis_server.get_connection_interval() ==
+                      conn_interval);
+                CHECK(redis_server.get_connection_attempts() ==
+                      expected_conn_attempts);
+
+                CHECK(redis_server.get_command_timeout() ==
+                      cmd_timeout);
+                CHECK(redis_server.get_command_interval() ==
+                      cmd_interval);
+                CHECK(redis_server.get_command_attempts() ==
+                      expected_cmd_attempts);
+            }
+        }
+    }
+    GIVEN("A negative value of " + std::string(CONN_TIMEOUT_ENV_VAR))
+    {
+        unset_all_env_vars();
+        setenv(CONN_TIMEOUT_ENV_VAR, "-1", true);
+        THEN("Constructor throws an exception")
+        {
+            CHECK_THROWS_AS(invoke_constructor(), Exception);
```

Comment:
Rather than checking for Exception, try to point to the specific type of SR Exception that will get thrown here